No active SSH connection found

╷
│ Warning: error reading the file on the remote host: no active SSH connection found
│
│   with cloudinit-drive.vm-gateway-cloudinit,
│ Error reading 'iso' file. The 'iso' file will be re-created.
╵
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-cloudinit-drive plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x7b4a8c]

goroutine 27 [running]:
github.com/pkg/sftp.(*Client).nextID(...)
        github.com/pkg/sftp@v1.13.5/client.go:264
github.com/pkg/sftp.(*Client).removeFile(0x0, {0xc00015e2c6, 0x2c})
        github.com/pkg/sftp@v1.13.5/client.go:734 +0x2c
github.com/pkg/sftp.(*Client).Remove(0xc0004c94a0?, {0xc00015e2c6, 0x2c})
        github.com/pkg/sftp@v1.13.5/client.go:717 +0x2c
terraform-provider-cloudinit-drive/cid.(*cloudInitDriveResource).Delete(0xc0003ae0c8, {0xd98d70, 0xc0005823f0}, {{{{0xd9d538, 0xc0001d6b70}, {0xb7eb00, 0xc000091230}}, {0xd9edc0, 0xc000136910}}, {{{0x0, ...}, ...}, ...}, ...}, ...)
        terraform-provider-cloudinit-drive/cid/resource_actions.go:114 +0x67a
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).DeleteResource(0xc00018f600, {0xd98d70, 0xc0005823f0}, 0xc0001c7378, 0xc000287328)
        github.com/hashicorp/terraform-plugin-framework@v1.3.2/internal/fwserver/server_deleteresource.go:90 +0x570
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0xc0001c7508?, {0xd98d70, 0xc0005823f0}, 0xc0003b2c30, 0xc000287508)
        github.com/hashicorp/terraform-plugin-framework@v1.3.2/internal/fwserver/server_applyresourcechange.go:79 +0x317
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0xc00018f600, {0xd98d70?, 0xc0005822a0?}, 0xc0003b2b90)
        github.com/hashicorp/terraform-plugin-framework@v1.3.2/internal/proto6server/server_applyresourcechange.go:55 +0x41a
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0001a15e0, {0xd98d70?, 0xc0004c98c0?}, 0xc000439b20)
        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/tf6server/server.go:819 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0xc3f4c0?, 0xc0001a15e0}, {0xd98d70, 0xc0004c98c0}, 0xc000439ab0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:422 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001e81e0, {0xd9d818, 0xc0003829c0}, 0xc000377e60, 0xc000338c00, 0x1263780, 0x0)
        google.golang.org/grpc@v1.56.1/server.go:1337 +0xdf3
google.golang.org/grpc.(*Server).handleStream(0xc0001e81e0, {0xd9d818, 0xc0003829c0}, 0xc000377e60, 0x0)
        google.golang.org/grpc@v1.56.1/server.go:1714 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.56.1/server.go:959 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.56.1/server.go:957 +0x18c

Error: The terraform-provider-cloudinit-drive plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/cid/resource_actions.go
+++ b/cid/resource_actions.go
@@ -37,7 +37,10 @@ func (r *cloudInitDriveResource) Read(ctx context.Context, req resource.ReadRequ
 	}
 
 	if strings.HasPrefix(state.DrivePath.ValueString(), "ssh://") {
-		r.client.Connect()
+		resp.Diagnostics.Append(r.client.Connect())
+		if resp.Diagnostics.HasError() {
+			return
+		}
 		defer r.client.Close()
 	}
 
@@ -108,7 +111,10 @@ func (r *cloudInitDriveResource) Delete(ctx context.Context, req resource.Delete
 
 	switch {
 	case strings.HasPrefix(state.DrivePath.ValueString(), "ssh://"):
-		r.client.Connect()
+		resp.Diagnostics.Append(r.client.Connect())
+		if resp.Diagnostics.HasError() {
+			return
+		}
 		defer r.client.Close()
 
 		err = r.client.scp.Remove(isoFile[6:])

